internal/constants/script-with-nmstatectl.go

```diff
@@ -0,0 +1,129 @@
+package constants
+
+// PreNetworkConfigScriptWithNmstatectl script runs on hosts before network manager service starts in order to apply
+// user's provided network configuration on the host.
+// If the user provides static network configuration, the network config files will be stored in directory
+// /etc/assisted/network in the following structure:
+// /etc/assisted/network/
+//
+//		+-- host1
+//		|      +--- *.yml
+//		|      +--- mac_interface.ini
+//		+-- host2
+//		      +--- *.yml
+//		      +--- mac_interface.ini
+//	 1. *.yml - files provided by the user with additional embedded captures to make it nmpolicy
+//	 2. mac_interface.ini - the file contains mapping of mac-address to logical interface name.
+//	    There are two usages for the file:
+//	      1. Map logical interface name to MAC Address of the host. The logical interface name is a
+//	         name provided by the user for the interface. It will be replaced by execute nmstatectl service with the
+//	         actual network interface name.
+//	      2. Identify if a host directory belongs to the current host by matching a MAC Address
+//	         from the mapping file with host network interfaces.
+//
+// Applying the network configuration of each host will be done by:
+//  1. Associate the current host with its matching hostX directories. The association will be done by
+//     matching host's mac addresses with those in mac_interface.ini.
+//  2. Replace logical interface name with the interface name as set on the host by nmstate
+//  3. The nmstate service generates keyfiles and places them in '/etc/NetworkManager/system-connections'.
+//     When the script is used with the minimal ISO, we copy these keyfiles from '/etc/NetworkManager/system-connections' to '/etc/coreos-firstboot-network'.
+const PreNetworkConfigScriptWithNmstatectl = `#!/bin/bash
+
+PATH_PREFIX=${PATH_PREFIX:=''}
+
+# The directory that contains nm config files of all nodes
+NM_CONFIG_DIR=${PATH_PREFIX}/etc/assisted/network
+MAC_NIC_MAPPING_FILE=mac_interface.ini
+NMSTATECTL=${PATH_PREFIX}/usr/bin/nmstatectl
+NMSTATE_DIR=${PATH_PREFIX}/etc/nmstate
+
+mkdir -p ${NMSTATE_DIR}
+
+if [[ ! -d "$NM_CONFIG_DIR" ]]
+then
+  echo "Error (exiting): Expected to find the directory $NM_CONFIG_DIR on the host but this was not present."
+  exit 0
+fi
+
+# A map of host mac addresses to interface names
+declare -A host_macs_to_hw_iface
+
+# Find destination directory based on ISO mode
+if [[ -f ${PATH_PREFIX}/etc/initrd-release ]]; then
+  ETC_NETWORK_MANAGER="${PATH_PREFIX}/etc/coreos-firstboot-network"
+else
+  ETC_NETWORK_MANAGER="${PATH_PREFIX}/etc/NetworkManager/system-connections"
+fi
```

Comment:
Some of this and some of the below script looks reused. Could we co-locate it somehow so that if/when we need to fix bugs we don't have to remember to fix them in two places?